src/main/scala/com/fulcrumgenomics/umi/ConsensusCallingIterator.scala

```diff
@@ -67,10 +67,11 @@ class ConsensusCallingIterator[ConsensusRead <: SimpleRead](sourceIterator: Iter
       groupingIterator.flatMap(caller.consensusReadsFromSamRecords)
     }
     else {
-      ParIterator(groupingIterator, threads=threads).flatMap { rs =>
+      ParIterator(groupingIterator, threads=threads, chunkSize=chunkSize).map { rs =>
         val caller = callers.get()
         caller.synchronized { caller.consensusReadsFromSamRecords(rs) }
-      }.toAsync(chunkSize * 8)
+      }.toAsync(chunkSize).flatten
+      // Flatten AFTER pulling through ParIterator to keep input chunks in phase with output
```

Comment:
If I understand this correctly, we previously got chunks out of order in some cases?
```suggestion
      // Flatten AFTER pulling through ParIterator to keep the iterator in a stable order
```